contracts/proxy/transparent/TransparentUpgradeableProxy.sol

```diff
@@ -175,6 +154,14 @@ contract TransparentUpgradeableProxy is ERC1967Proxy {
         return _getAdmin();
     }
 
+    /**
+     * @dev Makes sure the admin cannot access the fallback function. See {Proxy-_beforeFallback}.
+     */
+    function _beforeFallback() internal virtual override {
+        require(msg.sender != _getAdmin(), "TransparentUpgradeableProxy: admin cannot fallback to proxy target");
+        super._beforeFallback();
+    }
```

Comment:
Any particular concern for adding this?

Reply:
Note that this was here originally, it's being "added" only because #4154 is being reverted.